include/SFML/Window/WindowBase.hpp

```diff
@@ -202,24 +203,28 @@ class SFML_WINDOW_API WindowBase
     /// \brief Wait for an event and return it
     ///
     /// This function is blocking: if there's no pending event then
-    /// it will wait until an event is received. After this function
-    /// returns if no error occurred, the returned event will not be
-    /// empty. This function is typically used when you have a thread
-    /// that is dedicated to events handling: you want to make this
-    /// thread sleep as long as no new event is received.
+    /// it will wait until an event is received or until the provided
+    /// timeout elapses. After this function returns if no error nor
+    /// timeout occurred, the returned event will not be empty.
+    /// This function is typically used when you have a thread that is
+    /// dedicated to events handling: you want to make this thread sleep
+    /// as long as no new event is received. The default timeout value,
+    /// `Time::Zero`, means that the function will wait indefinitely.
     /// \code
     /// if (const auto event = window.waitEvent())
     /// {
     ///    // process event...
     /// }
     /// \endcode
     ///
-    /// \return The event
+    /// \param timeout Maximum time to wait (`Time::Zero` for infinite)
+    ///
+    /// \return The event; will be `Empty` (convertible to `false`) on timeout or if window was closed
     ///
     /// \see pollEvent
     ///
     ////////////////////////////////////////////////////////////
-    [[nodiscard]] Event waitEvent();
+    [[nodiscard]] Event waitEvent(Time timeout = Time::Zero);
```

Comment:
We shouldn't recommend Zero as default value, since we don't want threads to get stuck. I'd go with 5 or 10s.

```suggestion
    [[nodiscard]] Event waitEvent(Time timeout = seconds(5.f));
```

Reply:
Isn't that quite long? How about one second? Threads would use a loop anyway

Reply:
I even considered a higher timeout. You're not necessarily *expecting* events at a constant rate. If a user doesn't touch their keyboard, mouse, controller for some time, you may not get any events.

The timeout is really there as a ripcord in case we end up in a state where it's guaranteed that we'll never be receiving any events anymore.

So going down to 1s would IMHO trigger the timeout too often, while I think if you had to wait for the process to terminate 5s is about acceptable.

Reply:
> I even considered a higher timeout. You're not necessarily expecting events at a constant rate. If a user doesn't touch their keyboard, mouse, controller for some time, you may not get any events. [...]

But that's fine even with a shorter timeout, isn't it? The pattern would be:

```cpp
int main()
{
    std::atomic<bool> running{true};
    sf::RenderWindow rw(/* ... */);

    std::jthread eventHandlingThread{[&]
    {
        while (running)
            if (const sf::Event e = rw.waitEvent())
                // ...handle event...
    }};
    
    // ...do stuff that takes time...
    
    running.store(false);
}
```

The timeout only exists as a way of saying *"every X seconds, I'm going to check whether we can clean-up the event handling thread"*. 

I think that having a long default timeout of 5s or 10s might lead people to believe their app is hanging/freezing. I know I wouldn't have the patience to wait 10s and I'd start figuring out why my app is freezing (while in reality it isn't). 

> So going down to 1s would IMHO trigger the timeout too often

Triggering the timeout should not be an issue, what's your concern with it being triggered more often?

Reply:
I'd say zero/infinite is a sensible default timeout - it matches the existing behaviour and I'd say is typical for blocking functions when a timeout isn't specified, rather than picking some arbitrary value

Reply:
How about we just don't have a default at all? @eXpl0it3r

Reply:
No, I think Jonny is right. We're adding this for an edge case in some people's code. We shouldn't change the behavior for everyone because of that.

Reply:
I think this timeout should remain infinite by default. Let's not change the default behavior.

Reply:
Fair enough, I'll change the timeout to be consistent with the other functions here in preparation of a PR for https://github.com/SFML/SFML/pull/3094#discussion_r1637119797